### PR TITLE
Feature/cc 0100 adding pagination update test config

### PR DESCRIPTION
**Pull request by GitHubEason**
Add pagination
Screen Shot 2022-07-16 at 4 23 17 pm
n

### CODE REVIEW

File: src/common/dto/pagination-query.dto.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { IsOptional, IsPositive } from "class-validator";
+export class PaginationQueryDto {
+  @IsOptional()
+  @IsPositive()
+  limit: number;
+
+  @IsOptional()
+  @IsPositive()
+  offset: number;
+}
```
Comment (Jzeva) [inline src/common/dto/pagination-query.dto.ts:10]: I am not sure whether it is a good idea to add an independent directory "dto" in the common directory when other modules will put their own dto in their own directory. I think maybe it should be put into a folder like "utils".

File: src/main.ts [hunk 2]
```diff
@@ -7,7 +7,14 @@ async function bootstrap() {
   const app = await NestFactory.create<NestExpressApplication>(AppModule);
   app.enableCors();
   app.set("trust proxy", 1);
-  app.useGlobalPipes(new ValidationPipe({ whitelist: true }));
+  app.useGlobalPipes(
+    new ValidationPipe({
+      whitelist: true,
+      transformOptions: {
+        enableImplicitConversion: true,
+      },
```
Comment (Jzeva) [inline src/main.ts:15]: I am a bit curious about this part. Could you please expain it? I didn't find it in the Nestjs documentation but just some not implicit examples on the Internet. I will keep doing research on the Internet but I still expect your answer.

**Comment by Jzeva** [comment]
I noticed that you added pagination for all modules. Thanks for your efforts. I think maybe you can add some comments or documentation for it so that everyone who will use the pagination in front end or back end can easily apply your function to their code.

File: src/main.ts (hunk 2 above)
Comment (GitHubEason) [reply src/main.ts:15]: This one for pagnation

**Review by Fuchih — APPROVED** [review]

**Review by jennawan — APPROVED** [review]

**Pull request merged**